updated most ms

superDeano opened the pull request.

superDeano committed:
imported all modules into intellij

superDeano committed:
added retrofit

tests for testing the values
and interfaces for ms communications

superDeano committed:
making use of retrofit

superDeano committed:
implemented apis from other ms to be used

superDeano committed:
fixed some errors in api declaration

Also implemented some logic for interacting with filenames

superDeano committed:
remove unused api

Also fixed a bug

superDeano committed:
Updated devices ms

MS can now save deviceId to family account
retrieve list of devices for user or family account
add/remove devices
check ms health with respect to db connection
talk to user ms to validate users

superDeano committed:
updated gitignore and added main class to run the device ms

superDeano committed:
added ability for the notif ms to talk to device ms

to retrieve list of topics for a user

superDeano committed:
refactored name

superDeano committed:
deleted some code and removed some unnecessary import

superDeano committed:
updated info for app.properties

superDeano committed:
new feature for device ms

If a user admin deletes his account, all the information about his devices will also be deleted

superDeano committed:
removed application.yml and new features for users ms

user ms will be able to communicate to device and notif ms to share information

superDeano committed:
media ms has ability to delete all files for a deleted family account

superDeano committed:
user ms can now finally delete accounts properly

can also talk to other ms if need be

superDeano committed:
new api to retrieve the family account for the device

superDeano committed:
the media ms can now talk to device ms to retrieve fam acc to save file

superDeano committed:
fixed error in test

superDeano committed:
make sure to provide the list of devices to be deleted to notif ms

superDeano committed:
new notif api to delete all topics

superDeano committed:
slightly better at dealing with null pointer exceptions for user ms

superDeano committed:
forgot try catch there too

superDeano committed:
better way to get the environment variables into the spring app

superDeano committed:
proceeding to catch nullpointerexceptions 

and better deal with environment variables for notifications ms

superDeano committed:
small improvements

superDeano committed:
better micro service health check

superDeano committed:
updated device ms to better deal with npr exceptions

also updated unit test to create docker image
also update app.prop

superDeano committed:
better app.prop for building image

superDeano committed:
java doc

BigBen03 commented on Back End/devices/src/main/java/com/theelite/devices/model/Account.java:
```diff
@@ -16,5 +16,5 @@
     @Id
     private UUID accountId;
-    private UUID[] deviceIds;
+    private String[] deviceIds;
```
so we dropping uuid? or you just taking an array of uuid strings?

superDeano committed:
make sure to wait to delete topic

superDeano committed:
Features for User ms

- Added GsonConverter for retrofit
- Fixed urls in the interfaces
- Now saves the new family account

superDeano committed:
added parametrized call to prevent errors

superDeano committed:
added json converter and removed / in api def

superDeano committed:
added gson converter and updated apis for media ms

superDeano committed:
updated ms to use gson converter and apis

superDeano committed:
when someone needs to delete the account, 

the family account will then be deleted

superDeano committed:
fixed a bug where retrofit would not parse properly

also fixed a null pointer when retrieving list of device ids when there are none

superDeano committed:
new api and bugs fixing for the device ms

superDeano committed:
fixed a bug where json would not work

also fixed account model so they have the same info as with the other ms

superDeano committed:
add check to make sure that user requires password to delete his account

superDeano committed:
fixed a bug where the password would not be compared accurately

superDeano committed:
fixes the json bug

superDeano committed:
actually pings the db to test for app health

and implicitly ignore an error

superDeano committed:
Refactoring, fixing some null pointer exceptions and removed 

Amazon Web Services errors

superDeano committed:
fixes json bug with retrofit in notif ms

superDeano committed:
removed no longer useful tests

superDeano committed:
fixed api when talking to device

also made sure that cannot publish if topic does not exists

superDeano committed:
to ignore Paw files
